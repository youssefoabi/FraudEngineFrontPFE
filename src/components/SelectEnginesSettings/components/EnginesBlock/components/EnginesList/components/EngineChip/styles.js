import { doveGray, eucalyptus, shamrock, silver, white } from '@cdiscount/ui-colors';

export default function style() {
  return {
    baseStyle: {
      borderRadius: 5,
      color: white,
      marginLeft: 5,
      marginTop: 10,
      height: 25,
    },
    statusEnableChip: {
      backgroundColor: shamrock,
      '&:hover': {
        backgroundColor: shamrock,
      },
    },
    statusEnableCurrentChip: {
      backgroundColor: eucalyptus,
      '&:hover': {
        backgroundColor: eucalyptus,
      },
    },
    statusDisableChip: {
      backgroundColor: silver,
      '&:hover': {
        backgroundColor: silver,
      },
    },
    statusDisableCurrentChip: {
      backgroundColor: doveGray,
      '&:hover': {
        backgroundColor: doveGray,
      },
    },
  };
}
