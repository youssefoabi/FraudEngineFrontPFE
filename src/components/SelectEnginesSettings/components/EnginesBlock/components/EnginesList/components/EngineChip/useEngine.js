import { dissoc, pathOr, prop } from 'ramda';

const useEngine = ({ engineId, handleChange, enginesGroup }) => {
  const { engines: enginesGroupEngines, currentEngine } = enginesGroup || {};
  const engine = pathOr({}, [engineId], enginesGroupEngines);
  const idCurrentEngine = pathOr({}, ['engineId'], currentEngine);

  const onDelete = index => {
    handleChange({ id: 'engines', value: dissoc(index, enginesGroupEngines) });
    if (index === idCurrentEngine) handleChange({ id: 'currentEngine', value: null });
  };

  const onClick = clickedEngine => handleChange({ id: 'currentEngine', value: clickedEngine });

  const isEnabled = prop('isEnable', engine);

  return {
    onDelete,
    onClick,
    engine,
    isEnabled,
    idCurrentEngine,
  };
};

export default useEngine;
