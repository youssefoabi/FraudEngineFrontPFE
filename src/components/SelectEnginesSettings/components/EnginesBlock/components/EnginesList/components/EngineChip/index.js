import { func, number, shape } from 'prop-types';
import { isNilOrEmpty } from 'ramda-adjunct';
import { prop } from 'ramda';
import { useStyles } from '@cdiscount/backoffice-ui';
import clsx from 'clsx';
import MuiChip from '@material-ui/core/Chip';
import React, { useState } from 'react';

import DeleteEngineConfirmationModal from '../DeleteEngineConfirmationModal';
import styles from './styles';
import useEngine from './useEngine';

const EngineChip = ({ engineId, handleChange, enginesGroup }) => {
  const { onDelete, onClick, engine, isEnabled, idCurrentEngine } = useEngine({
    handleChange,
    enginesGroup,
    engineId,
  });
  const [deletionModalOpen, setDeletionModalOpen] = useState(false);
  const isCurrentEngine = engineId === idCurrentEngine;

  const toggleDeletionModal = () => {
    setDeletionModalOpen(!deletionModalOpen);
  };

  const classes = useStyles(styles);
  if (isNilOrEmpty(engine)) return null;

  return (
    <>
      <DeleteEngineConfirmationModal
        engineName={prop('engineName', engine)}
        confirm={() => onDelete(prop('engineId', engine))}
        open={deletionModalOpen}
        toggle={toggleDeletionModal}
      />
      <MuiChip
        label={prop('engineName', engine)}
        className={clsx(classes.baseStyle, {
          [classes.statusEnableCurrentChip]: isEnabled && isCurrentEngine,
          [classes.statusEnableChip]: isEnabled && !isCurrentEngine,
          [classes.statusDisableCurrentChip]: !isEnabled && isCurrentEngine,
          [classes.statusDisableChip]: !isEnabled && !isCurrentEngine,
        })}
        onDelete={() => toggleDeletionModal()}
        onClick={() => onClick(engine)}
      />
    </>
  );
};

EngineChip.propTypes = {
  enginesGroup: shape({}),
  engineId: number,
  handleChange: func,
};

EngineChip.defaultProps = {
  enginesGroup: null,
  engineId: null,
  handleChange: Function.Prototype,
};

export default EngineChip;
