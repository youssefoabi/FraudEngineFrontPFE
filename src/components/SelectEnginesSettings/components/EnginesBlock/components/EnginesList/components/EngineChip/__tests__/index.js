import { itRendersAllMutations } from '@cdiscount/jest-utils';

import EngineChip from '../';

jest.resetModules();

jest.mock('../useEngine', () => ({
  __esModule: true,
  default: jest
    .fn()
    .mockReturnValueOnce({})
    .mockReturnValueOnce({
      engine: { engineName: 'engine', engineId: 1 },
      onDelete: jest.fn(),
      onClick: jest.fn(),
      isEnabled: true,
    })
    .mockReturnValueOnce({
      engine: null,
    })
    .mockReturnValueOnce({
      engine: {},
    }),
  namedExport: 'useEngine',
}));

const mutations = [
  { name: 'without props' },
  { name: 'with props' },
  { name: 'with null engine' },
  { name: 'with empty engine' },
];

describe(EngineChip.name, () => {
  itRendersAllMutations(EngineChip, mutations);
});
