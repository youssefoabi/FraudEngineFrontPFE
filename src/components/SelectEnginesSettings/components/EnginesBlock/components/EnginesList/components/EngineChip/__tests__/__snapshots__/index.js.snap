// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineChip EngineChip renders correctly with empty engine 1`] = `""`;

exports[`EngineChip EngineChip renders correctly with null engine 1`] = `""`;

exports[`EngineChip EngineChip renders correctly with props 1`] = `
<Fragment>
  <DeleteEngineConfirmationModal
    confirm={[Function]}
    engineName="engine"
    open={false}
    toggle={[Function]}
  />
  <WithStyles(ForwardRef(Chip))
    className="makeStyles-baseStyle-6 makeStyles-statusEnableChip-7"
    label="engine"
    onClick={[Function]}
    onDelete={[Function]}
  />
</Fragment>
`;

exports[`EngineChip EngineChip renders correctly without props 1`] = `""`;
